### DISCUSSION TITLE
Announcement - v.1.15

### DISCUSSION THREAD

**Discussion by kevinpapst** [kimai/kimai#2768] [General]
[Full Changelog](https://github.com/kevinpapst/kimai2/compare/1.14.3...1.15)

**Implemented enhancements:**

- I would like to record an activity or description at the end of the time recording. Would it be possible? [\#2724](https://github.com/kevinpapst/kimai2/issues/2724)
- Return pagination related info in the api response [\#2720](https://github.com/kevinpapst/kimai2/issues/2720)
- Time entry popup: make a round clock or minutes in 5 min steps [\#2715](https://github.com/kevinpapst/kimai2/issues/2715)
- Cumbersome position of "End" button for active timesheet entry since Kimai 1.14 [\#2693](https://github.com/kevinpapst/kimai2/issues/2693)
- Add expense with two ore more categories [\#2666](https://github.com/kevinpapst/kimai2/issues/2666)
- DragAndDrop 10 RecentActivities configurable [\#2646](https://github.com/kevinpapst/kimai2/issues/2646)
- Not applicable tax case  [\#2592](https://github.com/kevinpapst/kimai2/issues/2592)
- Wrong version number for 1.14.3 [\#2582](https://github.com/kevinpapst/kimai2/issues/2582)
- Optionally hide Username/Password and Login button [\#2458](https://github.com/kevinpapst/kimai2/issues/2458)
- Instant Time Record [\#2215](https://github.com/kevinpapst/kimai2/issues/2215)
- Explain missing time period in timesheet export [\#2091](https://github.com/kevinpapst/kimai2/issues/2091)
- store search in session [\#2735](https://github.com/kevinpapst/kimai2/pull/2735) ([kevinpapst](https://github.com/kevinpapst))
- datatable sorting [\#2734](https://github.com/kevinpapst/kimai2/pull/2734) ([kevinpapst](https://github.com/kevinpapst))
- Added search modal for timesheet export [\#2728](https://github.com/kevinpapst/kimai2/pull/2728) ([kevinpapst](https://github.com/kevinpapst))
- Login screen UI [\#2711](https://github.com/kevinpapst/kimai2/pull/2711) ([kevinpapst](https://github.com/kevinpapst))
- Fixes for 1.15 [\#2704](https://github.com/kevinpapst/kimai2/pull/2704) ([kevinpapst](https://github.com/kevinpapst))
- Support multiple teamleads in each team [\#2702](https://github.com/kevinpapst/kimai2/pull/2702) ([kevinpapst](https://github.com/kevinpapst))
- move supported languages logic to service [\#2701](https://github.com/kevinpapst/kimai2/pull/2701) ([kevinpapst](https://github.com/kevinpapst))
- Classical cleanup PR [\#2700](https://github.com/kevinpapst/kimai2/pull/2700) ([kevinpapst](https://github.com/kevinpapst))
- improved play and stop button if only one record is allowed [\#2692](https://github.com/kevinpapst/kimai2/pull/2692) ([kevinpapst](https://github.com/kevinpapst))
- allow to merge multiple Kimai 1 installations [\#2691](https://github.com/kevinpapst/kimai2/pull/2691) ([kevinpapst](https://github.com/kevinpapst))
- allow 65535 character in meta fields [\#2690](https://github.com/kevinpapst/kimai2/pull/2690) ([kevinpapst](https://github.com/kevinpapst))
- monthly budgets and report [\#2684](https://github.com/kevinpapst/kimai2/pull/2684) ([kevinpapst](https://github.com/kevinpapst))
- added account number field to user [\#2671](https://github.com/kevinpapst/kimai2/pull/2671) ([kevinpapst](https://github.com/kevinpapst))
- Bugfixes [\#2669](https://github.com/kevinpapst/kimai2/pull/2669) ([kevinpapst](https://github.com/kevinpapst))
- better readability for drag and drop boxes [\#2659](https://github.com/kevinpapst/kimai2/pull/2659) ([kevinpapst](https://github.com/kevinpapst))
- improve permission checks for report access [\#2658](https://github.com/kevinpapst/kimai2/pull/2658) ([kevinpapst](https://github.com/kevinpapst))
- Configurable print export [\#2657](https://github.com/kevinpapst/kimai2/pull/2657) ([kevinpapst](https://github.com/kevinpapst))
- Project detail report [\#2651](https://github.com/kevinpapst/kimai2/pull/2651) ([kevinpapst](https://github.com/kevinpapst))
- refactor configurations [\#2626](https://github.com/kevinpapst/kimai2/pull/2626) ([kevinpapst](https://github.com/kevinpapst))
- Improve color choice and name validation [\#2622](https://github.com/kevinpapst/kimai2/pull/2622) ([kevinpapst](https://github.com/kevinpapst))
- added setting to limit the maximum length of a timesheet record [\#2612](https://github.com/kevinpapst/kimai2/pull/2612) ([kevinpapst](https://github.com/kevinpapst))
- removed soft\_limit setting [\#2611](https://github.com/kevinpapst/kimai2/pull/2611) ([kevinpapst](https://github.com/kevinpapst))
- Refactor authentication system [\#2602](https://github.com/kevinpapst/kimai2/pull/2602) ([kevinpapst](https://github.com/kevinpapst))
- allow to configure timezone for the lockdown period [\#2593](https://github.com/kevinpapst/kimai2/pull/2593) ([kevinpapst](https://github.com/kevinpapst))
- Lint container [\#2591](https://github.com/kevinpapst/kimai2/pull/2591) ([kevinpapst](https://github.com/kevinpapst))
- feat: adds support for setting description by http params [\#2580](https://github.com/kevinpapst/kimai2/pull/2580) ([tobybatch](https://github.com/tobybatch))
- limit amount of items in calendar drag and drop boxes [\#2291](https://github.com/kevinpapst/kimai2/pull/2291) ([kevinpapst](https://github.com/kevinpapst))
- Support for PHP 8 [\#2158](https://github.com/kevinpapst/kimai2/pull/2158) ([kevinpapst](https://github.com/kevinpapst))
- Add a calendar configuration event [\#2698](https://github.com/kevinpapst/kimai2/pull/2698) ([Jelle-S](https://github.com/Jelle-S))
- customer-name as part of the invoice number [\#2640](https://github.com/kevinpapst/kimai2/pull/2640) ([kevinpapst](https://github.com/kevinpapst))
- Translations update from Weblate [\#2739](https://github.com/kevinpapst/kimai2/pull/2739) ([weblate](https://github.com/weblate))
- Translations update from Weblate [\#2672](https://github.com/kevinpapst/kimai2/pull/2672) ([weblate](https://github.com/weblate))
- Translations update from Weblate [\#2665](https://github.com/kevinpapst/kimai2/pull/2665) ([weblate](https://github.com/weblate))
- Translations update from Weblate [\#2613](https://github.com/kevinpapst/kimai2/pull/2613) ([weblate](https://github.com/weblate))
- Translations update from Weblate [\#2577](https://github.com/kevinpapst/kimai2/pull/2577) ([weblate](https://github.com/weblate))
- Translations update from Weblate [\#2717](https://github.com/kevinpapst/kimai2/pull/2717) ([weblate](https://github.com/weblate))
- Translations update from Weblate [\#2661](https://github.com/kevinpapst/kimai2/pull/2661) ([weblate](https://github.com/weblate))

**Fixed bugs:**

- Bug in activities API [\#2765](https://github.com/kevinpapst/kimai2/issues/2765)
- No update to 1.14.3 it stops by 1.14.1 [\#2759](https://github.com/kevinpapst/kimai2/issues/2759)
- Kimai2 Migrations conflict with MySQL `ANSI_QUOTES` mode [\#2751](https://github.com/kevinpapst/kimai2/issues/2751)
- Exported XLS - cell format [\#2744](https://github.com/kevinpapst/kimai2/issues/2744)
- Changing public folder path [\#2742](https://github.com/kevinpapst/kimai2/issues/2742)
- copy entries not working if overlapping timentries are disabled [\#2731](https://github.com/kevinpapst/kimai2/issues/2731)
- update problem [\#2706](https://github.com/kevinpapst/kimai2/issues/2706)
- Tag colour defaults to grey on editing [\#2649](https://github.com/kevinpapst/kimai2/issues/2649)
- Version reporting mismatch on upgrade: `1.14.1` or `1.14.3`? [\#2648](https://github.com/kevinpapst/kimai2/issues/2648)
- Calendar is cut off at the bottom of the browser [\#2609](https://github.com/kevinpapst/kimai2/issues/2609)
- Reassigning activity to other customer/project does not adapt timesheet entries [\#2576](https://github.com/kevinpapst/kimai2/issues/2576)
- CSV importer adds only first missing customer, assigning all other projects to that new customer [\#2479](https://github.com/kevinpapst/kimai2/issues/2479)
- using the same tag multiple times  in csv not resolved [\#2318](https://github.com/kevinpapst/kimai2/issues/2318)
- Small color bug with projects [\#2200](https://github.com/kevinpapst/kimai2/issues/2200)
- Api fixes [\#2766](https://github.com/kevinpapst/kimai2/pull/2766) ([kevinpapst](https://github.com/kevinpapst))
- fix duplicating timesheet if allow overlapping is deactivated [\#2732](https://github.com/kevinpapst/kimai2/pull/2732) ([kevinpapst](https://github.com/kevinpapst))
- prevent email or username becoming non-unique [\#2730](https://github.com/kevinpapst/kimai2/pull/2730) ([kevinpapst](https://github.com/kevinpapst))
- fix team edit with deactivated users [\#2716](https://github.com/kevinpapst/kimai2/pull/2716) ([kevinpapst](https://github.com/kevinpapst))
- fix user stats in profile for first year [\#2712](https://github.com/kevinpapst/kimai2/pull/2712) ([kevinpapst](https://github.com/kevinpapst))
- Ldap [\#2689](https://github.com/kevinpapst/kimai2/pull/2689) ([kevinpapst](https://github.com/kevinpapst))
- LDAP: fixing issues for re-authenticating users [\#2681](https://github.com/kevinpapst/kimai2/pull/2681) ([kevinpapst](https://github.com/kevinpapst))
- make sure that timezone is properly validated [\#2663](https://github.com/kevinpapst/kimai2/pull/2663) ([kevinpapst](https://github.com/kevinpapst))
- bugfixes and improvements [\#2660](https://github.com/kevinpapst/kimai2/pull/2660) ([kevinpapst](https://github.com/kevinpapst))
- Bugfixes [\#2655](https://github.com/kevinpapst/kimai2/pull/2655) ([kevinpapst](https://github.com/kevinpapst))
- fix url under certain environments [\#2621](https://github.com/kevinpapst/kimai2/pull/2621) ([kevinpapst](https://github.com/kevinpapst))
- fix datepicker appears out of screen [\#2610](https://github.com/kevinpapst/kimai2/pull/2610) ([kevinpapst](https://github.com/kevinpapst))

Thanks to [Mohammed Al Oraimi](https://twitter.com/ixSly) and David Woodhouse from [Aegis9](https://www.aegis9.com.au) for reporting possible CSV attacks: called `Dynamic Data Exchange` or `DDE payload` attacks, actually a problem in Microsoft Excel. Do not rely on CSV for data exchange if you use Excel. This is not exclusively for Kimai, but applies for all software products that support user input and CSV export.

<hr /><em>This discussion was created from the release <a href='https://github.com/kevinpapst/kimai2/releases/tag/1.15'>1.15</a>.</em>

**Comment by leuedaniel** [comment]
What about
Add expense with two ore more categories #2666
there is nothing done if i see it right?

**Reply by kevinpapst** [reply]
Correct. But again: this is Kimai repository and has nothing to do with the plugin. Please don't mix them. You have access to the expense repository and I answered there.

**Comment by kevinpapst** [comment]
Feedback so far: 
- Database problem while updating: https://github.com/kevinpapst/kimai2/issues/2769 (configuration issue in installation)
- LDAP is not working: https://github.com/kevinpapst/kimai2/issues/2771 (configuration issue in installation)
- cache:clear failed - missing files in vendor: https://github.com/kevinpapst/kimai2/issues/2774 (BC break: fos_user config was removed)
- https://github.com/kevinpapst/kimai2/discussions/2783 (new config was added)
